shared/src/main/scala/com/cibo/evilplot/plot/Histogram.scala

```diff
@@ -117,11 +117,13 @@ object Histogram {
         // Scaling the bars would show the correct histogram as long as no axis is displayed.  However, if
         // an axis is display, we would end up showing the wrong values. Thus, we clip if the y boundary is
         // fixed, otherwise we scale to make it look pretty.
-        val points = binningFunction(data, plot.xbounds, binCount)
+        
+        val dataBounds = Bounds.get(data) getOrElse plot.xbounds
```

Comment:
This produces strange behavior when `plot.xbounds.range < Bounds.get(data).get.range`. For example, for `data` containing 1000 samples from a standard normal distribution, `Histogram(data).xbounds(0, 5)` produces
![hist-1000-standard](https://user-images.githubusercontent.com/14116434/46415381-27176f80-c6f3-11e8-90a4-3cbdcce18925.png)
If, instead, we do
```
val filteredData = data.filter(plot.xbounds.isInBounds)
val dataBounds = Bounds.get(filteredData).getOrElse(plot.xbounds)
val points = binningFunction(filteredData, dataBounds, binCount)
```
things look more the way I would expect
![hist-1000-standard](https://user-images.githubusercontent.com/14116434/46415435-4b734c00-c6f3-11e8-9f65-6611dacada75.png)

Reply:
Also, looking at both the above plots, it seems like some bars are floating a pixel or two above the x-axis. That bug definitely precedes these changes, but it's probably worth our looking into, too (not necessarily in this PR).